Assets/HoloToolkit/Utilities/Scripts/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,143 @@
+    {
+        [Tooltip("Checking enables SetFocusPointForFrame to set the stabilization plane.")]
+        public bool SetStabilizationPlane = true;
+        [Tooltip("Lerp speed when moving focus point closer.")]
+        public float LerpStabilizationPlanePowerCloser = 4.0f;
+        [Tooltip("Lerp speed when moving focus point farther away.")]
+        public float LerpStabilizationPlanePowerFarther = 7.0f;
+
+        [Tooltip("Used to temporarily override the location of the stabilization plane.")]
+        public Transform TargetOverride;
+
+        [Tooltip("Keeps track of position-based velocity for the target object.")]
+        public bool TrackVelocity = false;
+
+        [Tooltip("Use the GazeManager class to set the plane to the gazed upon hologram.")]
+        public bool UseGazeManager = true;
+
+        [Tooltip("Default distance to set plane if plane is gaze-locked.")]
+        public float DefaultPlaneDistance = 2.0f;
+
+        [Tooltip("Visualize the plane at runtime.")]
+        public bool DrawGizmos = false;
+
+        // Used for rendering plane gizmo.
+        private Vector3 planePosition;
+
+        private float currentPlaneDistance = 4.0f;
+
+        private Vector3 previousPosition;
+
+        private const float FOCUS_POINT_FRAMERATE = 60.0f;
+
+        /// <summary>
+        /// Updates the focus point for every frame after all objects have finished moving.
+        /// </summary>
+        private void LateUpdate()
+        {
+            if (SetStabilizationPlane && Camera.main != null)
+            {
+                if (TargetOverride != null)
+                {
+                    ConfigureTransformOverridePlane();
+                }
+                else if (UseGazeManager && GazeManager.Instance != null)
+                {
+                    ConfigureGazeManagerPlane();
+                }
+                else
+                {
+                    ConfigureFixedDistancePlane();
+                }
+
+#if UNITY_EDITOR
+                if (DrawGizmos)
+                {
+                    OnDrawGizmos();
+                }
+#endif
+            }
+        }
+
+        private void ConfigureTransformOverridePlane()
+        {
+            planePosition = TargetOverride.position;
+
+            Vector3 velocity = Vector3.zero;
+            if (TrackVelocity)
+            {
+                velocity = UpdateVelocity();
+            }
+            
+            // Place the plane at the desired depth in front of the camera and billboard it to the camera.
+            HolographicSettings.SetFocusPointForFrame(TargetOverride.position, -Camera.main.transform.forward, velocity);
+        }
+
+        private void ConfigureGazeManagerPlane()
+        {
+            Vector3 gazeOrigin = Camera.main.transform.position;
+            Vector3 gazeDirection = Camera.main.transform.forward;
+            float lastHitDistance = GazeManager.Instance.HitInfo.distance;
```

Comment:
Assets/HoloToolkit/Utilities/Scripts/StabilizationPlaneModifier.cs(92,19): warning CS0219: The variable `lastHitDistance' is assigned but its value is never used